frontend/src/lib/components/UnitPicker/CustomUnitModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { FilterType } from '~/types'
+import { RefCallback, useEffect, useState } from 'react'
+import { LemonModal } from 'lib/components/LemonModal'
+import { LemonButton } from 'lib/components/LemonButton'
+import { PureField } from 'lib/forms/Field'
+import { capitalizeFirstLetter } from 'lib/utils'
+import { LemonInput } from 'lib/components/LemonInput/LemonInput'
+import { HandleUnitChange } from 'lib/components/UnitPicker/UnitPicker'
+
+function chooseFormativeElementValue(
+    formativeElement: 'prefix' | 'postfix' | null,
+    filters: Partial<FilterType>
+): string {
+    if (formativeElement === 'prefix') {
+        return filters.aggregation_axis_prefix || ''
+    }
+
+    if (formativeElement === 'postfix') {
+        return filters.aggregation_axis_postfix || ''
+    }
+
+    return ''
+}
+
+export function CustomUnitModal({
+    isOpen,
+    onSave,
+    formativeElement,
+    filters,
+    onClose,
+    overlayRef,
+}: {
+    isOpen: boolean
+    onSave: (hx: HandleUnitChange) => void
+    formativeElement: 'prefix' | 'postfix' | null
+    filters: Partial<FilterType>
+    onClose: () => void
+    overlayRef: RefCallback<HTMLDivElement>
+}): JSX.Element | null {
+    const [localFormativeElementValue, setLocalFormativeElementValue] = useState<string>(
+        chooseFormativeElementValue(formativeElement, filters)
+    )
+    useEffect(() => {
+        setLocalFormativeElementValue(chooseFormativeElementValue(formativeElement, filters))
+    }, [formativeElement])
+
+    if (formativeElement === null) {
+        return null
+    }
+
+    return (
+        <LemonModal
+            overlayRef={overlayRef}
+            isOpen={isOpen}
+            onClose={onClose}
+            bringToFront={true}
+            title={`Custom ${formativeElement}`}
+            footer={
+                <>
+                    <LemonButton
+                        type="primary"
+                        onClick={() => onSave({ [formativeElement]: localFormativeElementValue })}
+                    >
+                        Save
```

Comment:
In the case of this modal I'd suggest "Apply" as the more relevant verb, since "Save" could imply saving the insight, which does not happen here

Reply:
💯